content/articles/rocket-lander-in-elm.md

```diff
@@ -0,0 +1,198 @@
+Title: Experience report: Rocket lander in Elm
+Date: 2015-02-15
+Short_summary: We build a small game in the programming language Elm.
+Category: Dev
+Authors: Tom
+
+*Reading time: ~10 minutes.*
+
+[Play the demo (desktop only, sorry!)](/rocket-lander-in-elm-extra/ship.html)
+
+In a comedy contest for the funniest language alive JavaScript would
+win hands-down. For that reason we're investigating various new
+compile-to-JavaScript languages. In this post we're looking at
+[Elm](http://elm-lang.org/).
+
+Why is JavaScript funny? Consider the following cases:
+
+```js
+> "hi" * 0
+NaN
+> "2" * 0
+0
+> "2" + 0
+"20"
+> null - undefined
+NaN
+```
+
+[The famous Wat video](https://www.destroyallsoftware.com/talks/wat)
+has more examples after the brief part about Ruby.
+
+People noticed this of course, and to fix these issues they started
+developing new languages that compile to JavaScript. Some well-known
+ones are [CoffeeScript](http://coffeescript.org/),
+[TypeScript](http://www.typescriptlang.org/) and
+[Dart](https://www.dartlang.org/).
+
+I have spent a reasonable amount of time with all of the above,
+writing actual production code. I consider all of them an improvement
+over the very low bar that is JavaScript, but none of them get me
+excited.
+
+Picking e.g. TypeScript: It is high-ceremony to define interfaces, and
+the type-safety only helps if a sufficiently large amount of code is
+type-annotated. It's like a phase change: Below a certain threshold
+one might as well just omit all annotations because they help so
+little.
```

Comment:
Are you making an argument, or are you saying I should add something like "Luckily type information already exists for many third party libraries" ?